.github/workflows/release.yml

```diff
@@ -1,158 +1,222 @@
 name: release
 
+on:
+  workflow_dispatch:
+    inputs:
+      dry_run:
+        description: "Run the release without actually releasing bits"
+        type: boolean
+        default: true
+      crate:
+        description: "The crate to release. (containerd-shim-wasmtime)"
+        type: string
+      version:
+        description: "The version of the crate to release. (e.g., 1.2.3)"
+        type: string
+
 concurrency:
-  group: release-${{ github.workflow }}-${{ github.head_ref || github.ref }}
+  group: release-${{ github.workflow }}-${{ inputs.crate }}-${{ inputs.version }}
   cancel-in-progress: true
 
-on:
-  push:
-    tags:
-      - '**/v[0-9]+.[0-9]+.*'
-
 env:
   CARGO_TERM_COLOR: always
 
 jobs:
-  parse:
-    runs-on: ubuntu-latest
-    name: Parse ref
+  pre-release:
+    name: pre-release checks
+    runs-on: "ubuntu-latest"
     outputs:
-      crate: ${{ steps.parse.outputs.crate }}
-      version: ${{ steps.parse.outputs.version }}
-      runtime: ${{ steps.parse.outputs.runtime }}
+      crate: ${{ inputs.crate }}
+      runtime: ${{ steps.runtime_sub.outputs.runtime }}
+      version: ${{ inputs.version }}
+      ### is_shim is a string, not a boolean, so use: is_shim == 'true'
+      is_shim: ${{ steps.runtime_sub.outputs.is_shim }}
     steps:
-      - uses: actions/checkout@v3
+      - name: Fail if branch is not main
+        if: github.event_name == 'workflow_dispatch' && github.ref != 'refs/heads/main'
+        run: |
+          echo "This workflow should not be triggered with workflow_dispatch on a branch other than main"
+          exit 1
+
+      - uses: actions/checkout@v4
+      ### Determine the name of the runtime and if it is a binary release or crates.io
+      - name: verify version input
+        uses: actions/github-script@v7
         with:
-          ref: ${{ github.ref }}
-      - id: parse
-        name: Parse ref
-        shell: bash
-        run: ./scripts/parse_ref.sh ${{ github.ref }} >> ${GITHUB_OUTPUT}
+          script: |
+            const version = '${{ inputs.version }}';
+            if(!version.match(/^[0-9]+.[0-9]+.*/)) {
```

Comment:
I don't think this sanity check account for the full semantic versioning check. For example, "1.0" seems to be able to pass this check, no? 

The suggested Regex for semantic versioning check is: https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string

Reply:
this is really needed any more we ended up matching it to the crate file: https://github.com/containerd/runwasi/blob/fc7aa53b0f58f1945afdc46bc29ceb7b5bd307d5/.github/workflows/release.yml#L79.  We could remove it or add it to a PR check with suggested regular expression

Since the PR needs to bump the version, maybe a check with something like https://github.com/obi1kenobi/cargo-semver-checks or that regex would be helpful